Migrate beyla documentation for OBI (#7399)
open-telemetry/opentelemetry.io: content/en/docs/zero-code/obi/configure/service-discovery.md

| `exclude_instrument`                                                                                             | Specify selection criteria for excluding services from being instrumented. Useful for avoiding instrumentation of services typically found in observability environments. Refer to the [exclude services from instrumentation](#exclude-services-from-instrumentation) section for details.                                            | list of objects | (unset)                                                                                                                                                       |

> | `default_exclude_instrument`                                                                                     | Disables instrumentation of OBI itself, Grafana Alloy, the OpenTelemetry Collector, and other observability components. Set to empty to allow OBI to instrument itself and these other components. Refer to the [default exclude services from instrumentation](#default-exclude-services-from-instrumentation) section for details. . | list of objects | Path: `{*beyla,*alloy,*prometheus-config-reloader,*ebpf-instrument,*otelcol,*otelcol-contrib,*otelcol-contrib[!/]*}` and certain Kubernetes system namespaces |

Reviewer: that's Ok!

> | `default_exclude_instrument`                                                                                     | Disables instrumentation of OBI itself, Grafana Alloy, the OpenTelemetry Collector, and other observability components. Set to empty to allow OBI to instrument itself and these other components. Refer to the [default exclude services from instrumentation](#default-exclude-services-from-instrumentation) section for details. . | list of objects | Path: `{*beyla,*alloy,*prometheus-config-reloader,*ebpf-instrument,*otelcol,*otelcol-contrib,*otelcol-contrib[!/]*}` and certain Kubernetes system namespaces |

Author: applied locally, will push later

| `exclude_otel_instrumented_services_span_metrics`<br>`OTEL_EBPF_EXCLUDE_OTEL_INSTRUMENTED_SERVICES_SPAN_METRICS` | Disables OBI span metric/service graph metric generation of services already instrumented with OpenTelemetry. Refer to the [exclude instrumented services](#exclude-otel-instrumented-services) section for details.                                                                                                    | boolean         | false                                                                                                                                                         |

## Discovery services

You can override the service name, namespace, and other configurations per
service type.

| YAML                   | Description                                                                                                                              | Type                     | Default                  |
| ---------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- | ------------------------ | ------------------------ |

> | `name`                 | Defines a name for the matching instrumented service. Refer to [name](#name).                                                            | string                   | (see description)        |

Author: removed

> | `name`                 | Defines a name for the matching instrumented service. Refer to [name](#name).                                                            | string                   | (see description)        |
> | `namespace`            | Defines a namespace for the matching instrumented service. Refer to [namespace](#namespace).                                             | string                   | (empty or K8s namespace) |

Author: ```suggestion
```

the exported metrics and traces.

This option is deprecated, as multiple matches for the same `instrument` entry
mean multiple services share the same name. Refer to the
[override service name and namespace](#override-service-name-and-namespace)
section to enable automatic configuration of service name and namespace from
diverse metadata sources.

If you don't set this property, OBI uses the following properties, in order of
precedence:

- If Kubernetes is enabled:
  1. The name of the Deployment that runs the instrumented process, if any
  2. The name of the ReplicaSet, DaemonSet, or StatefulSet that runs the
     instrumented process, if any
  3. The name of the Pod that runs the instrumented process
- If Kubernetes isn't enabled:
  1. The name of the process executable file

If multiple processes match the service selection criteria, the metrics and
traces for all the instances might share the same service name. For example,
when multiple instrumented processes run under the same Deployment, or have the
same executable name. In that case, the reported `instance` attribute lets you
differentiate the different instances of the service.

> ### Namespace

Author: removed

must match all the selector properties.

### K8s Pod labels

This selector property limits the instrumentation to the applications running in
the Pods with labels matching the provided value as glob.

If you specify other selectors in the same `instrument` entry, the processes
must match all the selector properties.

For example:

```yaml
discovery:
  instrument:
    - k8s_namespace: frontend
      k8s_pod_labels:
        instrument: obi
```

The preceding example discovers all Pods in the `frontend` namespace that have a
label `instrument` with a value that matches the glob `obi`.

### K8s Pod annotations

This selector property limits the instrumentation to the applications running in
the Pods with annotations matching the provided value as glob.

If you specify other selectors in the same `instrument` entry, the processes
must match all the selector properties.

For example:

```yaml
discovery:
  instrument:
    - k8s_namespace: backend
      k8s_pod_annotations:
        obi.instrument: 'true'
```

The preceding example discovers all Pods in the `backend` namespace that have an
annotation `obi.instrument` with a value that matches the glob `true`.

> ## Survey mode

Author: removed

> ## Survey mode
> 
> In survey mode, OBI only performs service discovery and detects the programming
> language of each service, but doesn't instrument any discovered services.
> 
> OBI writes the discovered information from survey mode to a metric called
> `survey_info`, which uses the same attributes as the `target_info` metric. The
> Prometheus exporter creates this metric based on the OpenTelemetry metric
> resource attributes. You can use survey mode to build external automated
> instrumentation solutions. For example, you can use the `survey_info` metric to
> list available instrumentation targets and choose which ones to instrument.
> 
> Configure the `survey` section exactly like the `instrument` section. For more
> details, see the [discovery services section](#discovery-services) of this
> document.
> 

Author: ```suggestion
```

## Exclude services from instrumentation

The `exclude_instrument` section lets you specify selection criteria for
excluding services from being instrumented. It follows the same definition
format as described in the [discovery services](#discovery-services) section of
this document.

This option helps you avoid instrumenting services typically found in
observability environments. For example, use this option to exclude
instrumenting Prometheus.

## Default exclude services from instrumentation

> The `default_exclude_instrument` section disables instrumentation of OBI itself
> (self-instrumentation), as well as Grafana Alloy and the OpenTelemetry
> Collector. It also disables instrumentation of various Kubernetes system

Author: ```suggestion
The `default_exclude_instrument` section disables instrumentation of OBI itself
(self-instrumentation), as well the OpenTelemetry Collector and other observability components. 
It also disables instrumentation of various Kubernetes system
```

Author: > Ok! (I don't have rights to accept your suggestion)

Another todo, we need to give @open-telemetry/ebpf-instrumentation-approvers permissions on this repo

Author: working on that.

namespaces to reduce the overall cost of metric generation. The following
section contains all excluded components:

> - Excluded services by `exe_path`: `*beyla`, `*alloy`, `*ebpf-instrument`,

Author: ```suggestion
- Excluded services by `exe_path`: `*beyla`, `*alloy`, `*ebpf-instrument`,
```

do we updated `*beyla*` here with `*obi*` or `*otel-ebpf*`?
